### PR TITLE
refactor: Add agw GitHub Actions Workflow

### PR DESCRIPTION
**Pull request by jwagantall**
-Add initial agw GHA workflow.

This workflow is migrated from CircleCI.

Signed-off-by: Jessica Wagantall <jwagantall@linuxfoundation.org>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/agw-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          sudo unzip protoc3.zip -d protoc3
+          sudo mv protoc3/bin/protoc /bin/protoc
+          sudo chmod a+rx /bin/protoc
+          sudo mv protoc3/include/google /usr/include/
+          sudo chmod -R a+Xr /usr/include/google
+          sudo rm -rf protoc3.zip protoc3
+          sudo mkdir ${CODEGEN_ROOT}
+          sudo wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O $SWAGGER_CODEGEN_JAR
+          sudo chmod -R 755 $CODEGEN_ROOT
+          sudo make -C $MAGMA_ROOT/lte/gateway/python test_all
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Can we do the test command in another step?

File: .github/workflows/agw-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    paths:
+      - 'orc8r/**'
+      - 'lte/**'
+
+jobs:
+
+########## LTE-TEST ##########
+
+  lte-test:
+
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml:172]: Get rid of line returns except on line 2

File: .github/workflows/agw-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Export environment
+      run: |
+          echo "MAGMA_ROOT=${PWD}" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: define MAGMA ROOT with github variables

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
LGTM, minor things explained in comments

File: .github/workflows/agw-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          SLACK_TITLE: "Github action lte-test update failed"
+          SLACK_USERNAME: "AGW workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+      # Notify ci channel when push succeeds
+      - name: Notify success to slack
+        if: success() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: We don't need to send an alert when this job succeeds

File: .github/workflows/agw-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            sudo wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O $SWAGGER_CODEGEN_JAR
+            sudo chmod -R 755 $CODEGEN_ROOT
+      - name: Run test_all
+        run: |
+            sudo make -C $MAGMA_ROOT/lte/gateway/python test_all
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          files: /var/tmp/test_results
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Could you add a name to the artifact like this
`   - name: Publish Unit Test Results
        uses: EnricoMi/publish-unit-test-result-action@v1
        if: always()
        with:
          files: /var/tmp/test_results
          name: lte_test_results`

**Comment by quentinDERORY** [comment]
There is one step missing where we upload lte and orc8r test files to codecov: https://github.com/magma/magma/blob/2744b3073be441531262d5991991ee770bdb2e77/.circleci/config.yml#L1158

File: .github/workflows/agw-workflow.yml (hunk 5 above)
Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml]: disregard use other workflow to upload like we discussed on other PRs

File: .github/workflows/agw-workflow.yml (hunk 4 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

**Comment by jwagantall** [comment]
lte-test was missing codecov still.. let me add it..

File: .github/workflows/agw-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+      - name: Install libraries and dependencies
+        env:
+          PYTHON_BUILD: "${{ env.MAGMA_ROOT }}/build"
+          PIP_CACHE_HOME: "${{ env.MAGMA_ROOT }}/.pipcache"
+          MAGMA_DEV_MODE: 1
+          SKIP_SUDO_TESTS: 1
+          CODEGEN_ROOT: "${{ env.MAGMA_ROOT }}/.codegen"
+          SWAGGER_CODEGEN_JAR: "${{ env.MAGMA_ROOT }}/.codegen/swagger-codegen-cli.jar"
+        run: |
+            mkdir -p /var/tmp/test_results
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml:46]: missing https://github.com/magma/magma/blob/206c84171a3fdd477cb6290750cd967138c92c1f/.circleci/config.yml#L212

File: .github/workflows/agw-workflow.yml [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+            sudo chmod -R a+Xr /usr/include/google
+            sudo rm -rf protoc3.zip protoc3
+            sudo -E mkdir ${{ env.CODEGEN_ROOT }}
+            sudo -E wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O ${{ env.SWAGGER_CODEGEN_JAR }}
+            # MAGMA_ROOT needs to be exported so that python.py is able to recognize it
+            export MAGMA_ROOT=${{ env.MAGMA_ROOT }}
+            sudo -E chmod -R 755 ${{ env.CODEGEN_ROOT }}
+            echo "$MAGMA_ROOT"
+            echo "Run test_all"
+            sudo -E make -C ${{ env.MAGMA_ROOT }}/lte/gateway/python test_all
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Can you separate a bit this huge step into multiple.
ie, one for the packages, one for protoc3, one for swagger codegen and one for the test run

File: .github/workflows/agw-workflow.yml [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+      REVISION: "${{ github.sha }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run mme-clang-tidy
+        uses: nick-invision/retry@v2
+        # yamllint disable rule:line-length
+        with:
+          command: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build -t magma-mme-build -f Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: long option names

File: .github/workflows/agw-workflow.yml [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+      REVISION: "${{ github.sha }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run mme-clang-warnings
+        uses: nick-invision/retry@v2
+        # yamllint disable rule:line-length
+        with:
+          command: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build -t magma-mme-build -f Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml (hunk 6 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:46]: You are right!.. added

File: .github/workflows/agw-workflow.yml (hunk 8 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!

File: .github/workflows/agw-workflow.yml (hunk 9 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure

File: .github/workflows/agw-workflow.yml (hunk 7 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!

**Comment by jwagantall** [comment]
@quentinDERORY  thanks for your comments,, 

Also.. Do you want me to also add long options in all the docker run commands too?
I figured I better ask before I change them since they are a lot :)

**Comment by quentinDERORY** [comment]
> @quentinDERORY thanks for your comments,,
> 
> Also.. Do you want me to also add long options in all the docker run commands too?
> I figured I better ask before I change them since they are a lot :)

yes that's a lot of options but go ahead for all of them :)

File: .github/workflows/agw-workflow.yml [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+  mme-clang-warnings:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+      BRANCH: "${{ github.base_ref }}"
+      REVISION: "${{ github.sha }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run mme-clang-warnings
+        uses: nick-invision/retry@v2
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Avoid using this module, it does not work well...

Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!.. shall i use the manual retry? If so, how many retries?

Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml]: I think for this specific step, you don't need to retry. There was no manual retry in circle ci

File: .github/workflows/agw-workflow.yml [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+        # yamllint disable rule:line-length
+        with:
+          command: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma-mme-build --file Dockerfile.ubuntu20.04 ../../../../
+            docker run --env BRANCH=${{ env.BRANCH }} --env REVISION=${{ env.REVISION }} --volume ${{ env.MAGMA_ROOT }}:/magma --volume ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma --interactive magma-mme-build:latest /bin/bash --cpu-shares "cd /magma/lte/gateway;make clang_warning_oai_upload"
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: -c is a bash option and not the `--cpu-shares`  option
I don't think there is a long one for it.

File: .github/workflows/agw-workflow.yml [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+        env:
+          PYTHON_BUILD: "${{ env.MAGMA_ROOT }}/build"
+          PIP_CACHE_HOME: "${{ env.MAGMA_ROOT }}/.pipcache"
+          SWAGGER_CODEGEN_JAR: "${{ env.MAGMA_ROOT }}/.codegen/swagger-codegen-cli.jar"
+          MAGMA_DEV_MODE: 1
+          SKIP_SUDO_TESTS: 1
+        run: |
+            sudo -E make -C ${{ env.MAGMA_ROOT }}/lte/gateway/python test_all
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: You must upload the test artifacts, the other workflow will take care of publishing it

File: .github/workflows/agw-workflow.yml [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+  c-cpp-codecov:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run codecov
+        uses: nick-invision/retry@v2
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Let's not use this module, just use a regular run command

File: .github/workflows/agw-workflow.yml (hunk 11 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: oops.. thanks!.. corrected

File: .github/workflows/agw-workflow.yml (hunk 13 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!

File: .github/workflows/agw-workflow.yml (hunk 12 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: are they in a different path? Can you show me an example?

File: .github/workflows/agw-workflow.yml (hunk 10 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: I noticed I am using the retry module a lot :/
Shall I do just one single run? or a manual retry like we did in the previous workflow?

File: .github/workflows/agw-workflow.yml (hunk 12 above)
Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml]: https://github.com/magma/magma/blob/8d1d33482689185ec38e63d5c1823ec77ea9f473/.github/workflows/cloud-workflow.yml#L152 with the path `/var/tmp/test_results`

File: .github/workflows/agw-workflow.yml (hunk 10 above)
Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml]: just don't use it, but some circle ci step had a specific timeout.
For this override the timeout at the step level. refer here https://stackoverflow.com/questions/59073731/set-default-timeout-on-github-action-pipeline

File: .github/workflows/agw-workflow.yml (hunk 12 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: thank you!..

File: .github/workflows/agw-workflow.yml (hunk 10 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure.. let me update those.

File: .github/workflows/agw-workflow.yml [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+        run: |
+            mkdir -p /var/tmp/test_results
+            mkdir -p /var/tmp/codecovs
+            sudo apt-get update -y
+            sudo apt-get install -y libsystemd-dev pkg-config curl zip unzip
+            sudo apt-get install -y virtualenv python-babel python-dev build-essential python3-setuptools python-setuptools autogen autoconf libtool python3-apt python3-requests python3-pip python-protobuf
+            pip3 install setuptools==49.6.0
+      - name: Setup protoc3
+        run: |
+            sudo curl -Lfs https://github.com/google/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip -o protoc3.zip
```
Comment (tmdzk) [inline .github/workflows/agw-workflow.yml]: We need to use the new protobuf link

File: .github/workflows/agw-workflow.yml [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+            sudo unzip protoc3.zip -d protoc3
```
Comment (tmdzk) [inline .github/workflows/agw-workflow.yml]: Let's not use sudo and write everything in a location where current user can manage the file

File: .github/workflows/agw-workflow.yml [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - master
+    paths:
+      - 'orc8r/**'
+      - 'lte/**'
+  pull_request:
+    branches:
+      - master
```
Comment (tmdzk) [inline .github/workflows/agw-workflow.yml:13]: This should also be tested on PR to `v*` branches example v1.5 v1.6 branches

File: .github/workflows/agw-workflow.yml [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8.5'
+      - name: Install python3-aioeventlet from the magma apt repo
+        run: |
+            cd nms/app
+            # Install python3-aioeventlet from the magma apt repo
+            cat ${{ env.MAGMA_ROOT }}/orc8r/tools/ansible/roles/pkgrepo/files/jfrog.pub | sudo apt-key add -
+            echo "deb https://facebookconnectivity.jfrog.io/artifactory/list/dev-focal/ focal main" | sudo tee /etc/apt/sources.list.d/fbc.list
```
Comment (tmdzk) [inline .github/workflows/agw-workflow.yml]: let's use the artifactory.magmacore.org instead

File: .github/workflows/agw-workflow.yml [hunk 18]
```diff
@@ -0,0 +1,301 @@
...
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action lte-test update failed"
+          SLACK_USERNAME: "AGW workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+  session_manager_test:
```
Comment (tmdzk) [inline .github/workflows/agw-workflow.yml:97]: let's just add a space to separates the jobs in a same workflow. just for readability

**Review by tmdzk — DISMISSED** [review]
How does caching works on Github Action nodes? Is there a caching layer?

**Comment by jwagantall** [comment]
> caching layer

@tmdzk  you mean something like this?
https://github.com/marketplace/actions/docker-layer-caching

File: .github/workflows/agw-workflow.yml (hunk 14 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 15 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: I change it to how is currently now in circleci

File: .github/workflows/agw-workflow.yml (hunk 16 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:13]: sure! added.

File: .github/workflows/agw-workflow.yml (hunk 17 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: Sure!.. can you confirm the URL.. is it https://artifactory.magmacore.org:443/artifactory/debian/focal?

File: .github/workflows/agw-workflow.yml (hunk 18 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:97]: @quentinDERORY  I believe we used to have a new line before a new job but we decided to remove it. 
Shall we follow @tmdzk suggestion and add them back?

**Comment by jwagantall** [comment]
@quentinDERORY , I wonder if this error is becuase of running test_all in a separate step 
https://github.com/magma/magma/runs/3188804470?check_suite_focus=true

Do you think so?

File: .github/workflows/agw-workflow.yml (hunk 18 above)
Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml:97]: let's do it :)

**Comment by tmdzk** [comment]
> > caching layer
> 
> @tmdzk you mean something like this?
> https://github.com/marketplace/actions/docker-layer-caching

Yes that, let's enable it, especially on the builds jobs

**Comment by quentinDERORY** [comment]
> @quentinDERORY , I wonder if this error is becuase of running test_all in a separate step
> https://github.com/magma/magma/runs/3188804470?check_suite_focus=true
> 
> Do you think so?

I don't think so, the only missing thing on this step is the CODEGEN_ROOT environment variable.
Maybe set the PYTHONPATH to MAGMA_ROOT as it looks like these are import errors

File: .github/workflows/agw-workflow.yml (hunk 18 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:97]: done :)

**Comment by jwagantall** [comment]
> > @quentinDERORY , I wonder if this error is becuase of running test_all in a separate step
> > https://github.com/magma/magma/runs/3188804470?check_suite_focus=true
> > Do you think so?
> 
> I don't think so, the only missing thing on this step is the CODEGEN_ROOT environment variable.
> Maybe set the PYTHONPATH to MAGMA_ROOT as it looks like these are import errors

Thanks.. let me try .. 

Actually, @quentinDERORY, I did the changes but i got now issues with imports for another module:
https://github.com/magma/magma/runs/3196521705?check_suite_focus=true

Investigating this..

File: .github/workflows/agw-workflow.yml [hunk 19]
```diff
@@ -0,0 +1,321 @@
...
+  push:
+    branches:
+      - master
+    paths:
+      - 'orc8r/**'
+      - 'lte/**'
+  pull_request:
+    branches:
+      - master
+      - 'v*'
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Can you replace 'v*' with 'v1.*'

File: .github/workflows/agw-workflow.yml [hunk 20]
```diff
@@ -0,0 +1,321 @@
...
+        with:
+          files: '/var/tmp/codecovs/cover_lte.xml'
+          flags: lte-test
+      - uses: codecov/codecov-action@v1
+        with:
+          files: '/var/tmp/codecovs/cover_orc8r.xml'
+          flags: lte-test
+      # Notify ci channel when failing
+      # Plugin info: https://github.com/marketplace/actions/slack-notify
+      - name: Notify failure to slack
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml:85]: Can you use commit title like here https://github.com/magma/magma/pull/8340

File: .github/workflows/agw-workflow.yml (hunk 19 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!.. fixed

File: .github/workflows/agw-workflow.yml (hunk 20 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:85]: sure!.. let me update it

Comment (quentinDERORY) [reply .github/workflows/agw-workflow.yml:85]: You updated it with the old code, can you use the new code from the PR linked.

File: .github/workflows/agw-workflow.yml [hunk 21]
```diff
@@ -0,0 +1,369 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Run codecov
+        # yamllint disable rule:line-length
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma/c_cpp_build --file Dockerfile.ubuntu20.04 ../../../../
+            ci_env=`bash <(curl -s https://codecov.io/env)`
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Can you use $() instead of `

File: .github/workflows/agw-workflow.yml (hunk 20 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml:85]: yes i did!.. i am soooooooooo sorry .. hahahha fixing it

File: .github/workflows/agw-workflow.yml (hunk 21 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed!

File: .github/workflows/agw-workflow.yml [hunk 22]
```diff
@@ -0,0 +1,369 @@
...
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Build the base c/c++ container
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma/c_cpp_build --file Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: Why not just specify the path to the Docker file instead of having to cd and then reference the buildcontext as magma root.
TLDR: remove the cd and use the absolute path on Dockerfile.ubuntu20.04 and change the build context from ../../../ to .

File: .github/workflows/agw-workflow.yml [hunk 23]
```diff
@@ -0,0 +1,369 @@
...
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: same thing here

File: .github/workflows/agw-workflow.yml [hunk 24]
```diff
@@ -0,0 +1,369 @@
...
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml [hunk 25]
```diff
@@ -0,0 +1,369 @@
...
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml [hunk 26]
```diff
@@ -0,0 +1,369 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Run mme-clang-tidy
+        timeout-minutes: 40
+        # yamllint disable rule:line-length
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma-mme-build --file Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml [hunk 27]
```diff
@@ -0,0 +1,369 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Run mme-clang-warnings
+        timeout-minutes: 40
+        # yamllint disable rule:line-length
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma-mme-build --file Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml [hunk 28]
```diff
@@ -0,0 +1,369 @@
...
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - name: Run codecov
+        # yamllint disable rule:line-length
+        run: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma/c_cpp_build --file Dockerfile.ubuntu20.04 ../../../../
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/agw-workflow.yml (hunk 22 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: sure!.. let me do that.. 
I copied it from what the original job had

File: .github/workflows/agw-workflow.yml (hunk 23 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 24 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 25 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 26 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 27 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml (hunk 28 above)
Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: fixed

File: .github/workflows/agw-workflow.yml [hunk 29]
```diff
@@ -0,0 +1,362 @@
...
+      - name: Extract commit title
+        # yamllint enable
+        if: failure() && github.ref == 'refs/heads/master'
+        id: commit
+        run: |
+            str="$(jq '.head_commit.message' $GITHUB_EVENT_PATH)"    # get the head_commit message
+            echo ::set-output name=title::${str%%\\n*} | tr -d '"'
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: The module version should be @v2.2.0

Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: updating it..

File: .github/workflows/agw-workflow.yml [hunk 30]
```diff
@@ -0,0 +1,362 @@
...
+            make -C ${{ env.MAGMA_ROOT }}/lte/gateway/python test_all
+      - name: Upload Test Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit Test Results
+          path: /var/tmp/test_results
+      - uses: codecov/codecov-action@v1
+        with:
+          files: '/var/tmp/codecovs/cover_lte.xml'
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: You can specify multiple files if they are comma separated.

Comment (jwagantall) [reply .github/workflows/agw-workflow.yml]: you mean to combine it with cover_orc8r.xml? sure.. let me do that!

**Review by quentinDERORY — APPROVED** [review]
You can add the paths to the pull_request event and then it's ready to go! 👍

**Comment by jwagantall** [comment]
thank you @quentinDERORY ! you have been so great all this time! thank you for that!
